lte/gateway/python/integ_tests/s1aptests/test_attach_he_policy.py

```diff
@@ -0,0 +1,232 @@
+            "ip_proto": FlowMatch.IPPROTO_TCP,  # Protocol Type
+            "direction": FlowMatch.UPLINK,  # Direction
+        }
+
+        # Flow lists to be configured
+        flow_list1 = [
+            ulFlow1,
+        ]
+
+        # QoS
+        qos1 = {
+            "qci": 1,  # qci value [1 to 9]
+            "priority": 1,  # Range [0-255]
+            "max_req_bw_ul": 10000000,  # MAX bw Uplink
+            "max_req_bw_dl": 15000000,  # MAX bw Downlink
+            "gbr_ul": 1000000,  # GBR Uplink
+            "gbr_dl": 2000000,  # GBR Downlink
+            "arp_prio": 1,  # ARP priority
+            "pre_cap": 1,  # pre-emption capability
+            "pre_vul": 1,  # pre-emption vulnerability
+        }
+
+        policy_id1 = "internet"
+
+        # Now actually complete the attach
+        attach = self._s1ap_wrapper._s1_util.attach(
+            ue_id,
+            s1ap_types.tfwCmd.UE_END_TO_END_ATTACH_REQUEST,
+            s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND,
+            s1ap_types.ueAttachAccept_t,
+        )
+        addr = attach.esmInfo.pAddr.addrInfo
+        default_ip = ipaddress.ip_address(bytes(addr[:4]))
+
+        # Wait on EMM Information from MME
+        self._s1ap_wrapper._s1_util.receive_emm_info()
+
+        # Delay to ensure S1APTester sends attach complete before sending UE
+        # context release
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+
+        print(
+            "********************** Sending RAR for IMSI",
+            "".join([str(i) for i in req.imsi]),
+        )
+        he_domain1 = "192.168.128.1"
+        assert utils.count_record_of_imsi_to_domain(imsi, he_domain1) == 0
+        self._sessionManager_util.create_ReAuthRequest(
+            "IMSI" + "".join([str(i) for i in req.imsi]),
+            policy_id1,
+            flow_list1,
+            qos1,
+            he_urls=HeaderEnrichment(urls=[he_domain1]),
+        )
+
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_ACT_DED_BER_REQ.value
+        )
+        act_ded_ber_req_oai_apn = response.cast(
+            s1ap_types.UeActDedBearCtxtReq_t
+        )
+        self._s1ap_wrapper.sendActDedicatedBearerAccept(
+            req.ue_id, act_ded_ber_req_oai_apn.bearerId
+        )
+
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Send PDN Connectivity Request
+        apn = "ims"
+        self._s1ap_wrapper.sendPdnConnectivityReq(ue_id, apn)
+        # Receive PDN CONN RSP/Activate default EPS bearer context request
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_PDN_CONN_RSP_IND.value
+        )
+        act_def_bearer_req = response.cast(s1ap_types.uePdnConRsp_t)
+        addr = act_def_bearer_req.m.pdnInfo.pAddr.addrInfo
+        sec_ip = ipaddress.ip_address(bytes(addr[:4]))
```

Comment:
see my comment regarding the secondary PDN. these lines will not be needed if you remove the second PDN in the test.